core/client/pom.xml

```diff
@@ -32,157 +30,52 @@
-         http://alluxio.org/documentation/Configuring-Alluxio-with-GlusterFS.html
-         for how to use glusterfs with Alluxio. -->
-    <!-- <dependency> -->
-    <!--   <groupId>org.alluxio</groupId> -->
-    <!--   <artifactId>alluxio-underfs-glusterfs</artifactId> -->
-    <!--   <version>${project.version}</version> -->
-    <!--   <exclusions> -->
-    <!--     <exclusion> -->
-    <!--       <groupId>org.alluxio</groupId> -->
-    <!--       <artifactId>alluxio-core-common</artifactId> -->
-    <!--     </exclusion> -->
-    <!--   </exclusions> -->
-    <!-- </dependency> -->
-    <dependency>
-      <groupId>org.alluxio</groupId>
-      <artifactId>alluxio-underfs-s3</artifactId>
-      <version>${project.version}</version>
-      <exclusions>
-        <exclusion>
-          <groupId>org.alluxio</groupId>
-          <artifactId>alluxio-core-common</artifactId>
-        </exclusion>
-      </exclusions>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>annotations</artifactId>
     </dependency>
+
+    <!-- Other projects' test-jars -->
     <dependency>
       <groupId>org.alluxio</groupId>
-      <artifactId>alluxio-underfs-local</artifactId>
+      <artifactId>alluxio-core-common</artifactId>
       <version>${project.version}</version>
-      <exclusions>
-        <exclusion>
-          <groupId>org.alluxio</groupId>
-          <artifactId>alluxio-core-common</artifactId>
-        </exclusion>
-      </exclusions>
+      <type>test-jar</type>
+      <scope>test</scope>
     </dependency>
   </dependencies>
 
   <profiles>
-    <!-- Profile used to build alluxio client jar used for Apache Spark. This will be activated by using -Pspark-->
-    <!-- in the maven command line. -->
     <profile>
       <id>spark</id>
-      <activation>
-        <activeByDefault>false</activeByDefault>
-      </activation>
-      <!-- The dependencies are the same as the default dependencies, except that curator-client, curator-framework, -->
-      <!-- curator-recipies and hadoop-client are in scope provided, as they are provided by Apache Spark runtime. -->
       <dependencies>
-        <dependency>
-          <groupId>org.apache.curator</groupId>
-          <artifactId>curator-framework</artifactId>
-          <version>${apache.curator.version}</version>
-          <scope>provided</scope>
-        </dependency>
-        <dependency>
-          <groupId>org.apache.curator</groupId>
-          <artifactId>curator-client</artifactId>
-          <version>${apache.curator.version}</version>
-          <scope>provided</scope>
-        </dependency>
-        <dependency>
-          <groupId>org.apache.curator</groupId>
-          <artifactId>curator-recipes</artifactId>
-          <version>${apache.curator.version}</version>
-          <scope>provided</scope>
-        </dependency>
+        <!-- Override the dependency of hadoop-client even it is already in the spark profile in parent pom. -->
```

Comment:
I get it now, how about
`<!-- Override the dependency of hadoop-client even though it is already in the spark profile in parent pom. This is needed because the dependency is introduced in this file and has a higher priority than the parent profile -->`